frontend/src/pages/exploreApplication/EnableModal.tsx

```diff
@@ -54,15 +54,19 @@ const EnableModal: React.FC<EnableModalProps> = ({ selectedApp, shown, onClose }
   React.useEffect(() => {
     if (validationInProgress && validationStatus === EnableApplicationStatus.SUCCESS) {
       setValidationInProgress(false);
+      // TODO: Disable rule below temporarily. Refactor to notify the owner and avoid modifying the object directly.
+      /* eslint-disable no-param-reassign */
       selectedApp.spec.isEnabled = true;
       selectedApp.spec.shownOnEnabledPage = true;
+      /* eslint-enable no-param-reassign */
+
       onClose();
     }
     if (validationInProgress && validationStatus === EnableApplicationStatus.FAILED) {
       setValidationInProgress(false);
       setPostError(validationErrorMessage);
     }
-  }, [onClose, selectedApp.spec, validationErrorMessage, validationInProgress, validationStatus]);
+  }, [onClose, selectedApp, validationErrorMessage, validationInProgress, validationStatus]);
```

Comment:
Undo this change. While the logic in this file is fishy, we don't need to add unnecessary changes to this PR.